### DISCUSSION TITLE
Generic transports

### DISCUSSION THREAD

**Discussion by roshanrags** [actix/actix-web#3314] [Ideas]
It seems like TcpListener/TcpStream are pretty deeply embedded in the current structure. Are generic transports something you would be interested in? The goal would basically be so users can supply their own listeners and/or streams and have actix run on top of them. It would ideally also allow higher layers like the Server to be oblivious to Tcp and Uds transports for example instead of what seems like two different sets of functions today.

**Comment by roshanrags** [comment]
FWIW, axum seems to be moving in this direction - https://github.com/tokio-rs/axum/pull/2479

**Comment by robjtede** [comment]
I'm definitely interested in generic transports, it would help a lot with TLS. It's something I'd like to experiment with when my backlog is cleared. Until then, if there's any ideas for a design, I'd encourage a feature request issue to be opened.
